### ISSUE TITLE
Failure to parse tail remarks `--` with a prefixed whitespace and without a space after the double dashes

### ISSUE THREAD

**Issue by ronaldtse** [lutaml/expressir#132]
<img width="736" alt="Screenshot 2023-12-07 at 1 07 02 PM" src="https://github.com/lutaml/expressir/assets/11865/2085df6c-7804-49c1-8a0c-37b6a6a53c67">
<img width="760" alt="Screenshot 2023-12-07 at 1 07 10 PM" src="https://github.com/lutaml/expressir/assets/11865/95503559-d91b-4e5c-8e82-9ab8e90a7902">
<img width="741" alt="Screenshot 2023-12-07 at 1 07 26 PM" src="https://github.com/lutaml/expressir/assets/11865/106c41c0-2ab4-4b16-a0c7-1f81bd71f883">

This is a valid tail remark:
```
SCHEMA a_schema;--remark
```

Another valid one:
```
 --valid remark
SCHEMA--valid remark
```

This schema fails validation due to two lines. However, it is supposed to pass.

[geometric_model_schema.exp.zip](https://github.com/lutaml/expressir/files/13596885/geometric_model_schema.exp.zip)

**Comment by ronaldtse** [comment]
```ruby
require 'expressir'
require 'expressir/express/parser'
Expressir::Express::Parser.from_file('geometric_model_schema.exp')
=> see error raised
```

**Comment by ronaldtse** [comment]
Ping @maxirmx to see if you have time.

**Comment by maxirmx** [comment]
I guess the root cause is the precedence.  Definition of remarks is in the tail of grammar file (have lowest precedence) and it doed not seem to be correct.

**Comment by ronaldtse** [comment]
@maxirmx indeed the “mis-reference” of the remark tag is a problem, but the remark tag technically does not need to cause a crash?

It is important though to report that a remark tag points to an invalid object. 

In any case, empty remarks should not fail. 

And in any case, the space between the two hyphens and the first double quote should not need to be present.

**Comment by maxirmx** [comment]
This is the existing implementation (part 1)
-------------
This is remark ```body``` tagged by ```ref```
```
--"ref"  body
```
-------------
This  is remark ``` "ref"  body``` not tagged  because a tag definition assumes no space between ```--``` and ```"ref"```
```
-- "ref"  body
```

IMHO this matches the grammar

**Comment by maxirmx** [comment]
Addressed by #133
